### ISSUE TITLE
How to handle null values of `isic_4digit` column in `ei_activities` dataset?

### ISSUE THREAD

**Issue by kalashsinghal** [2DegreesInvesting/tiltIndicator#570]
@AnneSchoenauer There are null values as well in `isic_4digit` column which could create issues while assigning risk categories for benchmarks `isic_sec` and `unit_isic_sec` for `Emissions profile` and `Emissions profile Upstream`. Please let me know how to handle such cases! Thanks!

**Comment by AnneSchoenauer** [comment]
Hi @kalashsinghal to get a better understanding of the problem could you please show me some examples where there are the null cases? Thanks!

**Comment by kalashsinghal** [comment]
Please have a look at this `pstr_companies.csv` dataset:

![image](https://github.com/2DegreesInvesting/tiltIndicator/assets/125359076/766a26f9-b616-4214-b349-03939c86cc04)

Here all the `not available` isic_4digit values are converted to NAs by the new refactored code. These NA values could cause issues if we want to segregate products based on different hierarchies of ISIC sectors for Emissions profile and Emissions profile upstream indicators (as to be done through this [issue](https://github.com/2DegreesInvesting/tiltIndicator/issues/571)). I hope now you understand the problem with NAs in `isic_4digit` column. 

These NAs come from the `ecoinvent_complete.csv` dataset which is present here: https://github.com/2DegreesInvesting/NLPTiltMatch/blob/main/data/input/ecoinvent_complete.csv

Please let me know if you need more clarity on this! :)

**Comment by AnneSchoenauer** [comment]
@kalashsinghal thanks a lot - this becomes clearer now. I think what we would need to do is to not take the NAs into account when we create the benchmark. Most likely this would be needed to be implemented in the tiltIndicator package when the benchmark is calculated? So that Mauro knows taht there can be NAs. It is totally correct to have the NAs here.

**Comment by kalashsinghal** [comment]
@AnneSchoenauer We have a file named `ecoinvent_complete.csv` ([file](https://github.com/2DegreesInvesting/NLPTiltMatch/blob/main/data/input/ecoinvent_complete.csv)) which consists of these `not available` ISIC sectors. I am not completely sure from where this file comes from. I guess it came from the ecoinvent but I am not sure whether any data manipulation is being done or not. 

I have sent the csv file you asked by email. Please have a look at it and let me what you think! Thanks!

**Comment by AnneSchoenauer** [comment]
@kalashsinghal Thanks a lot for this. Could you please help me as I don't understand on how we derived at the tilt_sectors in those cases. Normally the tilt_sectors are derived with the mapping of our ep_product to  ecoinvent product. Those ecoinvent product normally have a ISIC code and this ISIC code is then mapped to tilt_sectors. However, it seems to be the case that in the csv file that you sent to me that the sectors are derived differently (as there is no ISIC). Is it done via the sector resolving? Do you know?

**Comment by kalashsinghal** [comment]
@AnneSchoenauer Yes you are right! I just checked that the `clustered` column from the sector resolving file is used to map `tilt_sector` and `tilt_subsector`. The `isic_tilt_mapper.csv` file did mapping of `tilt_sector` and `tilt_subsector` using `isis_4digit`, however sector resolving file is used to substitute the NA values in `tilt_sector` and `tilt_subsector` if they are not mapped by the `isic_tilt_mapper.csv`.

**Comment by kalashsinghal** [comment]
@AnneSchoenauer I think we should not assign `tilt_sector` and `tilt_subsector` where the `isic_4digit` is `not available`/NA. Do you agree?

**Comment by AnneSchoenauer** [comment]
@kalashsinghal I think this is not correct. I think we should just make sure that when calculating the benchmarks that Mauro knows that for the isic and isic_unit benchmarks taht there are some data points not availabe. For the tilt_sec and tilt_sec_unit benchmarks however, he is then taking all products into account that have a tilt_sec and a tilt_subsector regardless of if they also have an isic or isic_unit.

**Comment by AnneSchoenauer** [comment]
@kalashsinghal can you follow up with Mauro on this? He will work on the benchmarking issue so it would be great if you could follow up with a ticket on this.

**Comment by kalashsinghal** [comment]
@AnneSchoenauer I noticed `not available` values in `isic_4digit` column while writing tests. I came to the conclusion after discussion with you that its alright to have null values in the `isic_4digit` column. So, I will just convert `not available` values into `NA` values and then I don't think I need to inform Mauro about the null values because they were neglected in the code before me noticing this as well. Does it make sense?
